### PR TITLE
Added architecture s390x to jib maven build task

### LINKED ISSUE #2337: Enable receiver and dispatcher Docker images for s390x architecture
**Describe the bug**
The following Docker images do not support the s390x architecture:

- gcr.io/knative-releases/knative-kafka-broker-receiver
- gcr.io/knative-releases/knative-kafka-broker-dispatcher

 **Expected behavior**
The given Docker images are multi-arch and contain layers for the s390x platform in addition to x86_64 and aarch64.

 **To Reproduce**
Deploy knative-eventing on s390x (zSystems) platform.

 **Knative release version**
n/a

 **Additional context**
Any PR attempting to address this should be able to use the excellent groundwork done with PR https://github.com/knative-sandbox/eventing-kafka-broker/pull/1968

### PR DESCRIPTION
**Pull request by haubenr**
Signed-off-by: Dirk Haubenreisser <haubenr@de.ibm.com>

Fixes #2337 

## Proposed Changes

This PR enables the Docker image build for the dataplane components (including receiver and dispatcher) for the s390x architecture. For this it leverages jib's multi-arch Docker image build capabilities.

**Release Note**

```release-note
Add support s390x architecture.
```

### CODE REVIEW

File: data-plane/pom.xml [hunk 1]
```diff
@@ -397,7 +397,7 @@
             <!--  KinD e2e tests registry uses HTTP -->
             <allowInsecureRegistries>true</allowInsecureRegistries>
             <from>
-              <image>gcr.io/distroless/java17-debian11:nonroot</image>
+              <image>eclipse-temurin:17-jre-focal</image>
```
Comment (pierDipi) [inline data-plane/pom.xml:400]: Can we consider contributing to the distroless project to include s390x Java images?

There already have support for s390x in some images https://github.com/GoogleContainerTools/distroless/pull/612

Comment (pierDipi) [reply data-plane/pom.xml:400]: They have added arm64 recently https://github.com/GoogleContainerTools/distroless/pull/874 and we would just need to add s390x to the `ARCHITECTURES` array?
https://github.com/GoogleContainerTools/distroless/blob/b90f68384b5456e9ed6a37a1205d339fd431778b/java/BUILD#L10

Comment (haubenr) [reply data-plane/pom.xml:400]: Well, there's a few things - on one hand, working with bazel on s390x is somewhat of a nightmare... the last bazel version I managed to compile and run on s390x is 3.7.2 which might be too old for GCT/distroless. On the other hand it's not sufficient to just add 's390x' to the ARCHITECTURES array. A whole bunch of additions would have to be made:
- debian.bzl
- checksums.bzl
- busybox_archives.bzl
- debian_versions.bzl
- debian_packages.yaml
- ...

And there's no guarantee that all the required Debian packages for s390x to support Java exist. Trying to add s390x support to GCT/distroless might open a whole can of worms... I'd prefer to switch the JRE for dispatcher and receiver to Eclipse Temurin Docker images seeing they are supporting all relevant architectures already (at least from IBM's perspective there's ppc64le support for kafka eventing, too which is not that far off).

What are the objections against Eclipse Temurin Docker images? Is it licensing concerns? Or the fact that they're hosted on Docker Hub (where pull rate limits apply)? Or that there are no rootless images available?

**Comment by haubenr** [comment]
@pierDipi Fwiw, I managed to build bazel 3.7.2 for s390x and actually built the GCT Java Docker images using this PR: https://github.com/GoogleContainerTools/distroless/pull/1068

Not sure if this gains any traction in the GCT repo... despite that PR I would rather go with the Temurin images (just a gut feeling). But _if_ it gets merged we could stick with GCT.

**Comment by pierDipi** [comment]
Thanks for contributing to distroless as well!
I see we're very close to get distroless images merged https://github.com/GoogleContainerTools/distroless/pull/1068!

**Comment by haubenr** [comment]
The GCT distroless PR has been merged. I will check the resulting Java Docker images and if they contain the proper s390x layers I'll update this PR and go back from Temurin to GCT.

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

**Pull request merged**